### PR TITLE
[FIX] AMS mztab

### LINKED ISSUE #5543: Incomplete mzTab metadata for Metabolomics
Bildschirmfoto 2021-09-20 um 13 05 11

This is all we have. At least filenames should be there.

### PR DESCRIPTION
**Pull request by oliveralka**
# Description

Please include a summary of the change and which issue is fixed.

fixes #5543

# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp [hunk 1]
```diff
@@ -915,9 +921,15 @@ namespace OpenMS
     ConsensusMap::ColumnHeaders fd_map = cmap.getColumnHeaders();
     Size num_of_maps = fd_map.size();
 
+    // corresponding file locations
+    std::vector<String> file_locations;
+    for (Size i = 0; i < fd_map.size(); ++i)
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:926]: could use a range for loop

File: src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp [hunk 2]
```diff
@@ -868,33 +876,31 @@ namespace OpenMS
   {
     f.getPeptideIdentifications().resize(f.getPeptideIdentifications().size() + 1);
     f.getPeptideIdentifications().back().setIdentifier(search_engine_identifier);
-    for (std::vector<AccurateMassSearchResult>::const_iterator it_row  = amr.begin();
-         it_row != amr.end();
-         ++it_row)
+    for (const auto& result : amr)
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:879]:
```suggestion
    for (const AccurateMassSearchResult& result : amr)
```

File: src/tests/class_tests/openms/data/AccurateMassSearchEngine_output1.featureXML [hunk 3]
```diff
@@ -119,7 +120,7 @@
 			</PeptideIdentification>
 			<UserParam type="string" name="label" value="T141"/>
 			<UserParam type="int" name="num_of_masstraces" value="1"/>
-			<UserParam type="floatList" name="masstrace_intensity" value="[36310.164094606996514]"/>
+			<UserParam type="floatList" name="masstrace_intensity" value="[3.6310164094607e04]"/>
```
Comment (timosachsenberg) [inline src/tests/class_tests/openms/data/AccurateMassSearchEngine_output1.featureXML:123]: why is this 1/10000

Comment (oliveralka) [reply src/tests/class_tests/openms/data/AccurateMassSearchEngine_output1.featureXML:123]: Should be the same only in scientific.

**Comment by jpfeuffer** [comment]
Awesome. Quick. Did you test it with a consensusXML, e.g. after aligning?

**Comment by jpfeuffer** [comment]
Whoops the diff was too large. There it is. Looks good!

**Review by jpfeuffer — APPROVED** [review]

File: src/tests/class_tests/openms/data/AccurateMassSearchEngine_output1_consensusXML.mzTab [hunk 4]
```diff
@@ -14,41 +24,41 @@ MTD	study_variable[8]-description	Accurate mass search result file.
...
+SML	HMDB:HMDB14907	C27H41NO6	[H][C@@]12CC[C@](O)(C(=O)COC(=O)CN(CC)CC)[C@@]1(C)C[C@H](O)[C@@]1([H])[C@@]2([H])CCC2=CC(=O)CC[C@]12C	InChI=1S/C27H41NO6/c1-5-28(6-2)15-23(32)34-16-22(31)27(33)12-10-20-19-8-7-17-13-18(29)9-11-25(17,3)24(19)21(30)14-26(20,27)4/h13,19-21,24,30,33H,5-12,14-16H2,1-4H3/t19-,20-,21-,24+,25-,26-,27-/m0/s1	Hydrocortamate	539.31071622816296	539.309157845695722	1	921.389662323789025	null	null	HMDB	3.5	null	[, , AccurateMassSearch, ]	null	null	null	null	null	null	null	null	null	null	null	null	3.12282e06	0.0	0.0	6.58385e06	0.0	0.0	4.31776e06	0.0	0.0	8316.330078125	0.0	0.0	4.89429e06	0.0	0.0	4.96792e06	0.0	0.0	9657.2998046875	0.0	0.0	6282.509765625	0.0	0.0	0.0	0.0	0.0	2.5240599609375e04	0.0	0.0	2.889590218461454	M+CH3CN+Na;1+	-1	null	475.294946431467281	1
```
Comment (jpfeuffer) [inline src/tests/class_tests/openms/data/AccurateMassSearchEngine_output1_consensusXML.mzTab:28]: Hmm search_engine_score_ms_run is always null? Doesnt AMS have a score? E.g. massdiff?

**Review by jpfeuffer — CHANGES_REQUESTED** [review]

File: src/tests/class_tests/openms/data/AccurateMassSearchEngine_output1_consensusXML.mzTab (hunk 4 above)
Comment (oliveralka) [reply src/tests/class_tests/openms/data/AccurateMassSearchEngine_output1_consensusXML.mzTab:28]: Yes, currently "null" was used all the time, but I think using the mass error is a good idea. 
I guess, it is enough to use the mass error in ppm?

Comment (jpfeuffer) [reply src/tests/class_tests/openms/data/AccurateMassSearchEngine_output1_consensusXML.mzTab:28]: Yeah I guess ppm. Renaming makes sense, too, if no effort.

**Comment by oliveralka** [comment]
Added the "MassErrorPPMScore". 

It is now reported as "best_search_engine_score" in the MzTab.

FeatureXML has only one score per file and identification.

For consensusXML - in case of AccurateMassSearch - the same is true since the mz of the ConsensusFeature is used for identification. Which means that there is only one score for all files.

**Comment by jpfeuffer** [comment]
Ah, right.. ok fine.

**Review by jpfeuffer — APPROVED** [review]

**Pull request merged**